lambda/repository/collection_validation.py

```diff
@@ -0,0 +1,415 @@
+#   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+#   Licensed under the Apache License, Version 2.0 (the "License").
+#   You may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+"""Validation service for collection configurations."""
+
+import logging
+import re
+from typing import Any, Dict, List, Optional
+
+from models.domain_objects import (
+    ChunkingStrategy,
+    CollectionMetadata,
+    CreateCollectionRequest,
+    FixedChunkingStrategy,
+    UpdateCollectionRequest,
+)
+from repository.collection_repo import CollectionRepository
+from repository.vector_store_repo import VectorStoreRepository
+from utilities.validation import ValidationError
+
+logger = logging.getLogger(__name__)
+
+# Validation constants
+MAX_NAME_LENGTH = 100
+MAX_TAG_LENGTH = 50
+MAX_TAGS_COUNT = 50
+MIN_CHUNK_SIZE = 100
+MAX_CHUNK_SIZE = 10000
+NAME_PATTERN = re.compile(r"^[a-zA-Z0-9 _-]+$")
+TAG_PATTERN = re.compile(r"^[a-zA-Z0-9_-]+$")
+
+
+class CollectionValidationService:
```

Comment:
This whole class feels weird that it exists. I don't really get why these can't be pydantic validators.